### PR TITLE
Add UTRSBot notification scripts

### PR DESCRIPTION
**Pull request by dqwiki**
@TParis Could you write a description for the change log and for this issue?

I'll try and do a code review ASAP.

### CODE REVIEW

**Comment by TParis** [comment]
New feature: Applies a notification to a blocked user's talk page upon email verification
New feature: New "Blocking Admin" button automatically notifies blocking admins upon a request
New feature: New "WMF Staff" button emails community outreach and places ticket in HOLD queue
New feature: For IP appeals only, will request open proxy check when placed in OPP queue
Bug fix: Repaired "Autorized" language in ACP
Bug fix: Repaired missing [$lang] when moving ticket to OPP queue

File: public_html/appeal.php [hunk 1]
```diff
@@ -257,6 +259,17 @@
...
 				$appeal->setStatus(Appeal::$STATUS_ON_HOLD);
 				$log->addNewItem(SystemMessages::$log['StatusOnHold'][$lang], 1);
+				
+				//Notify the blocking admin, if asked for
+				if ($_GET['value'] == "adminhold") {
+					$bot = new UTRSBot();
+					$time = date('M d, Y H:i:s', time());
+				    $bot->notifyAdmin($appeal->getBlockingAdmin(), array($appeal->getID(), $time));	
+					$log->addNewItem(SystemMessages::$log['NotifiedAdmin'][$lang], 1);			
+				} elseif ($_GET['value'] == "adminhold") {
```
Comment (dqwiki) [inline public_html/appeal.php]: Would this not be WMF hold?

**Comment by dqwiki** [comment]
Ok, so several questions:

1) Is the omission of subst: for the templates on purpose?
2) Could we perhaps rename the onwiki templates to a uniform style vs:

` private $userTemplate		= "Unblock-utrs";`
`   private $adminTemplate		= "Unblock-UTRS-AdminNotify";`
`   private $oppTemplate			= "UTRS-OPP";`

to maybe something like UTRS-unblock, UTRS-AdminNotify, UTRS-OPP?
3) With how unblock-utrs is worded and coloured (like the normal awaiting review of an admin, and open) would it not be prudent to change it to the reviewed color and mark it as declined or accepted like normal unblock requests?
4) L269 of appeal.php seems to have the wrong if statement (comment made direct in diff)
5) We never should be throwing the direct IP from the "CU data" in the DB on to wiki. I know it's "protected" by other if statements, but routed through a function such as getCommonName() so we can ensure continual integrity of private data
6) In OPP deferral `$appeal->getCommonName()` duplicates `$appeal->getIP()` in theoretical functionality
7) When deferring to OPP fails because hasAccount is true, a message should be in place explaining why a CU needs to review this instead of a general error message about posting manually. The error message still could be used as a general failure
8) I don't follow what is happening or being checked here: https://github.com/UTRS/utrs/commit/45f3f8e214d3ea355de7d2f772ab8a5f5c1d2932#diff-b7f273a2c2eb9a486a8c5a03b50c115fR663
9) In reviewing the following theoretical case:
> Does reset to new allow posting of a hashed IP, revealing publicly a hash that could be used to confirm if it's an old IP or not, then violating data retention?
It would not violate data retention, and that is a non-issue here. But posting hashes of IPs on closed appeals...should be prohibited as they serve no cause. Maybe throw a quick check?

Awesome work so far though :D

File: public_html/appeal.php (hunk 1 above)
Comment (TParis) [inline public_html/appeal.php]: Yes, it would.

**Comment by TParis** [comment]
These changes should resolve your concerns & Salvrim!'s on-wiki concerns.  The template changes can be made on-wiki.

File: public_html/appeal.php [hunk 2]
```diff
@@ -264,9 +264,9 @@
 				if ($_GET['value'] == "adminhold") {
 					$bot = new UTRSBot();
 					$time = date('M d, Y H:i:s', time());
-				    $bot->notifyAdmin($appeal->getBlockingAdmin(), array($appeal->getID(), $time));	
+				    $bot->notifyAdmin($appeal->getCommonName(), array($appeal->getID(), $appeal->getCommonName(), $time));
```
Comment (dqwiki) [inline public_html/appeal.php]: Not that I really give a crap, but you still have redundancy here, where you could just make bot->notifyAdmin() force the admin's name into the template.

**Comment by dqwiki** [comment]
1 & 7 - Haven't been addressed.
3 - I haven't looked onwiki for such changes yet, cause it's late
6 - See my comment that I made recently, I could care less if that is actually fixed.

**Comment by TParis** [comment]
1) Subst was removed because the BRFA requested that we remove it.

**Comment by TParis** [comment]
3) This change should be made on-wiki and shouldn't affect the pull
6) I like the template vars to be separate from the rest.  More control and less chance of mishap
7) I'm not sure what you mean by this
10) I can incorporate #40 pretty easily, but that will make the other code redundant.

**Comment by TParis** [comment]
10) See 974be2fbe707ae1c281cfa4f9420f1adf79204f2

This change doesn't remove "blocking admin" from the new appeal form.  It only replaces the user input in the appeal object.  This hasn't been tested - at all.

**Comment by dqwiki** [comment]
Screw 7, my logic was flawed
3 will be deferred into a separate issue for a later date
1...i'll figure out if I actually want to pursue this onwiki, and is not needed here.

**Comment by dqwiki** [comment]
@TParis after fixing several issues, I have tested:

1) Account, account block
2) Account, IP block
3) IP block

Is there anything else testing needed?
Also please review and sign off on the latest changes

**Comment by dqwiki** [comment]
Since the changes were approved (I think) and the rest were bug fixes, some very critical, i'm pushing this to live.

**Pull request merged**